### PR TITLE
#168781655 - Get trips stats

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
Users view bookings

#### Description of Task to be completed?
- Create actions/reducers for getStats.
- Create component and route for stats.
- Create tests.

#### How should this be manually tested?
- Clone ft-trips-stats-168781655 branch
- run `npm i`
- run `npm run dev`
- Click on `Stats` in the `SearchBar`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168781655](https://www.pivotaltracker.com/story/show/168781655)

#### Screenshots (if appropriate)
Screen Shot 2020-01-09 at 17 57 23

#### Questions:
N/A

### CODE REVIEW

**Review by ngireric123 — APPROVED** [review]
LGTM

File: src/tests/actionsTests/requestsActionsTest.spec.js [hunk 1]
```diff
@@ -254,4 +254,57 @@ it('Should trigger SEARCH_REQUESTS_FAIL', async () => {
...
+    type: GET_STATS_SUCCESS
+  }];
+  store = mockStore({});
+  await store.dispatch(getStatsAction())
+    .then(async () => {
+      const calledActions = store.getActions();
+      expect(calledActions).toEqual(expectedActions);
+    });
+});
+
```
Comment (Cheza-Dzabala) [inline src/tests/actionsTests/requestsActionsTest.spec.js:283]: remove this line

**Review by Cheza-Dzabala — CHANGES_REQUESTED** [review]
Great job @nignanthomas however, I have a few changes I'd like to suggest.

1. Please remove the `stats button` from the `search bar` and place it above it. 
2. The `date range` inputs must be placed on a white background.
3. Please line up the `date range` with the results from a search

File: src/views/requests/RequestsStats.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+export class RequestsStats extends Component {
+  state = {
+    isLoading: false,
+    startDate: '',
+    endDate: '',
+    emptyParams: false,
+  };
+
+  // componentDidMount = async () => {
```
Comment (nakiwuge) [inline src/views/requests/RequestsStats.js]: remove commented code

File: src/views/requests/RequestsStats.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    const { props } = this;
+    const { startDate, endDate } = this.state;
+
+    if (!startDate || !endDate) {
+      this.setState({ emptyParams: true });
+    } else {
+      this.setState({ isLoading: true, emptyParams: false });
+
+      const statsParams = `?startDate=${startDate}&endDate=${endDate}`;
+      console.log('statsParams ===> ', statsParams);
```
Comment (nakiwuge) [inline src/views/requests/RequestsStats.js]: Remove console

File: src/views/requests/RequestsStats.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+              Enter a date range to get the stats...
+            </Col>
+          </Row>
+
+          <Row>
+            <Col xs={12} sm={6} md={4} lg={4}>
+              <Form.Group className="day-picker-custom">
+                <Form.Label>Start Date...</Form.Label>
+                <DayPickerInput className="day-picker" inputProps={{ style: { border: 0, outline: 0 } }} placeholder="YYYY-MM-DD" value={startDate} onDayChange={this.handleStartDate} dayPickerProps={{ disabledDays: { after: today } }} />
+                {/* <i className="fa fa-times reset-cross" aria-hidden="true" onClick={this.resetDate} /> */}
```
Comment (nakiwuge) [inline src/views/requests/RequestsStats.js]: Remove commented code

**Review by nakiwuge — CHANGES_REQUESTED** [review]
Good job @nignanthomas however kindly remove the stats button from the search form and either add it on the menu component or put it on the right aligned it with the Search By button.

**Review by Cheza-Dzabala — APPROVED** [review]
LGTM @nignanthomas

**Review by Cheza-Dzabala — APPROVED** [review]

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**